### PR TITLE
Standardizing Logging Format for Improved Middleware Integration

### LINKED ISSUE #2866: test_runner: use `log.info("%s", user_id)` instead of `log.info(f"{user_id}")`; use f-strings instead of `.format(`
Quoting https://github.com/neondatabase/neon/pull/2062/files#r917153823:

> This method works better for integration with various logging middleware. For example if you use Sentry it will be able do dedup events by taking into account only format string, not the parameters. If parameters are pre filled like you proposed then it wont be able to do so and will create separate tickets for every combination of user_id/request_id/whatever that came with the event. It is highly unlikely that we'll need to use something like Sentry in tests, but this is the way that I used to cook python logging just to adhere to least surprise principle

Similarly with `.format(` calls. We have quite a few of both interpolation ways.

While we are here, we may as well replace `.format` with f-strings.

Currently, we use f-strings a lot more frequently.

### PR DESCRIPTION
**Pull request by macdoos**
### Problem: (#2866)
The issue highlights a mismatch in string formatting methods for logging in the codebase, affecting integration with logging middleware like Sentry, which can't efficiently deduplicate events due to the current use of f-strings and `.format()`.

### Changes:
1. **Logging Format**: Shift from `log.info(f"{user_id}")` to `log.info("%s", user_id)` for better compatibility with logging middleware.
2. **String Formatting**: Replace all `.format()` instances with f-strings for consistency and modernization outside of logging contexts.

Closes #2866

## Checklist before requesting a review

- [x] I have performed a self-review of my code.

## Checklist before merging

- [x] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: test_runner/fixtures/broker.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def try_start(self):
             return
 
         listen_addr = self.listen_addr()
-        log.info(f'starting storage_broker to listen incoming connections at "{listen_addr}"')
+        log.info("%s", 'starting storage_broker to listen incoming connections at "{listen_addr}"')
```
Comment (bayandin) [inline test_runner/fixtures/broker.py:34]: It seems we should use an f-string for the second argument here

**Review by bayandin — CHANGES_REQUESTED** [review]
Thanks for the PR, @macdoos! 

f-strings look great, I've been trying to do it for a long time! 
Could you find an appropriate linter rule to ensure we keep the code consistent in the future? (we use `ruff`, so something https://docs.astral.sh/ruff/rules/ may fit)? 

Changes for logging are a bit more controversial for our use case, but I think we can merge it as well if we can find a rule to keep it consistent.

Also, please run `poetry run ruff check --fix .` and `poetry run ruff format .` to make the linter happy.

File: test_runner/fixtures/broker.py (hunk 1 above)
Comment (macdoos) [reply test_runner/fixtures/broker.py]: good catch, ty!

**Comment by macdoos** [comment]
@bayandin, I've added [UP032](https://docs.astral.sh/ruff/rules/f-string/) to the ruff as a linter rule. I've also run `poetry`, and all the checks have passed. This should be mergeable.

**Comment by jcsp** [comment]
Which middleware do we care about for logs from tests?  The description mentions sentry, but I don't think we rely on that for managing test logs?

If we're going to require developers to write a spurious `"%s",` before every log format string, there should be a really good reason.

**Comment by macdoos** [comment]
I'm not entirely sure about the use case here since I'm not fully aware of the purpose of these tests, given that it's my first PR. However, I thought standardizing the logging format, as described in #2866, would be a good start.

**Comment by macdoos** [comment]
@bayandin _Done!_

File: test_runner/regress/test_sharding.py [hunk 2]
```diff
@@ -327,21 +327,8 @@ def get_node_shard_counts(env: NeonEnv, tenant_ids):
...
-            for k, v in non_default_tenant_config.items():
-                assert config.effective_config[k] == v
-
-    # Validate pageserver state: expect every child shard to have an attached and secondary location
-    (total, attached) = get_node_shard_counts(env, tenant_ids=[tenant_id])
-    assert sum(attached.values()) == split_shard_count
-    assert sum(total.values()) == split_shard_count * 2
-    check_effective_tenant_config()
+    log.info(f"Shards after split: {shards_exist}")
+    assert len(shards_exist) == split_shard_count
```
Comment (bayandin) [inline test_runner/regress/test_sharding.py:331]: These changes look incorrect, we shouldn't ament the logic

File: test_runner/regress/test_sharding.py [hunk 3]
```diff
@@ -830,7 +817,7 @@ def assert_rolled_back(exclude_ps_id=None) -> None:
             locations = ps.http_client().tenant_list_locations()["tenant_shards"]
             for loc in locations:
                 tenant_shard_id = TenantShardId.parse(loc[0])
-                log.info(f"Shard {tenant_shard_id} seen on node {ps.id} in mode {loc[1]['mode']}")
+                log.info(f"Shard {tenant_shard_id} seen on node {ps.id}")
```
Comment (bayandin) [inline test_runner/regress/test_sharding.py:820]: I don't think we need to changes this message

**Review by bayandin — CHANGES_REQUESTED** [review]
Thanks again @macdoos!

I see some amount of unrelated changes, probably leftovers from merge conflicts and from changes in logs.

I would recommend to reset your branch from the main, then apply changes in `pyproject.toml` (add `"UP032"` rule), and then run

```sh
poetry run ruff check --fix .
poetry run ruff format .
```

This should automatically fix most of the cases.

**Comment by macdoos** [comment]
@bayandin, apologies for the multiple messages. I've just performed a reset and included the linter. Everything should be in order now!

**Review by bayandin — APPROVED** [review]
Great work @macdoos 🎉

**Pull request merged**